pandas/io/parsers.py

```diff
@@ -786,17 +786,25 @@ def read(self, nrows=None):
             return ret
 
         # May alter columns / col_dict
-        index, columns, col_dict = self._create_index(ret)
+        index, columns, mappings, col_dict = self._create_index(ret)
 
-        df = DataFrame(col_dict, columns=columns, index=index)
+        if not mappings:
```

Comment:
then do it in a while loop. Its the same idea, or you can hadnle parse_dates first. again this is disassocitive from the actual construction process. you can also handle with a dict if you have uniques, otherwise do somethign else if you have duplicates (but then again if you have duplicates we don't allow duplicate names in usecols or parsedates, so I think you can still do it).

ideally would just have a nice loop-ish thing do this.